include/verilated_random.h

```diff
@@ -231,6 +201,43 @@ class VlRandomizer final {
     // METHODS
     // Finds the next solution satisfying the constraints
     bool next(VlRNG& rngr);
+    template <typename T_Key>
+    typename std::enable_if<std::is_integral<T_Key>::value>::type
+    process_key(const T_Key& key, std::string& indexed_name, size_t& integral_index,
+                const std::string& base_name, size_t& idx_width) {
+        integral_index = static_cast<size_t>(key);
+        indexed_name = base_name + "[" + std::to_string(integral_index) + "]";
+        idx_width = sizeof(T_Key) * 8;
+    }
+    template <typename T_Key>
+    typename std::enable_if<std::is_same<T_Key, std::string>::value>::type
+    process_key(const T_Key& key, std::string& indexed_name, size_t& integral_index,
+                const std::string& base_name, size_t& idx_width) {
+        integral_index = string_to_integral(key);
+        indexed_name = base_name + "[" + std::to_string(integral_index) + "]";
+        idx_width = 64;  // 64-bit mask
+    }
+    template <typename T_Key>
+    typename std::enable_if<!std::is_integral<T_Key>::value
+                            && !std::is_same<T_Key, std::string>::value>::type
+    process_key(const T_Key& key, std::string& indexed_name, size_t& integral_index,
+                const std::string& base_name, size_t& idx_width) {
+        VL_FATAL_MT(__FILE__, __LINE__, "randomize",
+                    "Unsupported: Only integral and string index of associative array is "
+                    "supported currently.");
+    }
+    size_t string_to_integral(const std::string& str) {
+        size_t result = 0;
+        for (char c : str) {
+            result = (result << 8) | static_cast<size_t>(c);
+            result &= 0xFFFFFFFFFFFFFFFF;
```

Comment:
```suggestion
            result &= 0xFFFFFFFFFFFFFFFFULL;
```

But, I don't understand why this is needed, it seems a nop given it's a 64 bit variable.